File: camp/leetcode/jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        maxReach, leng =0, len(nums)
        if leng==1:
            return True
            
        for i in range(leng):
            if maxReach <i:
                return False
            
            maxReach= max(maxReach, nums[i]+i)
            if maxReach >=leng-1:
                return True

File: 55-jump-game/jump-game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let len = nums.length;
    if (len === 1) return true;
    
    let pos = len-1;
    for (let i = len-2; i>=0; i--) {
        if (i+nums[i] >= pos) {
            pos = i;
            if (i == 0) return true;
        }
    }
    return false;
};